ggshield/scan/scanner.py

```diff
@@ -68,15 +68,40 @@ def from_exception(exc: Exception) -> "Results":
         return Results(results=[], errors=[error])
 
 
-class ScanCollection(NamedTuple):
+class ScanCollection:
     id: str
     type: str
     results: Optional[Results] = None
-    scans: Optional[List["ScanCollection"]] = None  # type: ignore[misc]
+    scans: Optional[List["ScanCollection"]] = None
     iac_result: Optional[IaCScanResult] = None
     optional_header: Optional[str] = None  # To be printed in Text Output
     extra_info: Optional[Dict[str, str]] = None  # To be included in JSON Output
 
+    def __init__(
+        self,
+        id: str,
+        type: str,
+        results: Optional[Results] = None,
+        scans: Optional[List["ScanCollection"]] = None,
+        iac_result: Optional[IaCScanResult] = None,
+        optional_header: Optional[str] = None,
+        extra_info: Optional[Dict[str, str]] = None,
+    ):
+        self.id = id
+        self.type = type
+        self.results = results
+        self.scans = scans
+        self.iac_result = iac_result
+        self.optional_header = optional_header
+        self.extra_info = extra_info
+
+        self.has_new_secrets = self._init_has_new_secrets()
```

Comment:
This one can be a property: it's only used once.

Also, I was surprised by the names of `has_new_secrets` and `has_new_secrets_all_scans`. I expected `has_new_secrets` to be true for scans and sub-scans. Plus `has_new_secrets_all_scans` does not sound English to me (but I could be wrong on that 😉). I suggest renaming them like this:

- `has_new_secrets` → `current_scan_has_new_secrets`
- `has_new_secrets_all_scans` → `has_new_secrets`